src/consensus/aft/raft.h

```diff
@@ -193,7 +196,8 @@ namespace aft
       std::chrono::milliseconds election_timeout_,
       std::chrono::milliseconds view_change_timeout_,
       size_t sig_tx_interval_ = 0,
-      bool public_only_ = false) :
+      bool public_only_ = false,
+      bool join_as_learner_ = false) :
```

Comment:
I would mildly prefer if we passed initial_state here, because we could perhaps kill off the force_become_leader() that way, but I'm happy for that to be a separate PR, since this is purely cosmetic.

Reply:
No problem, that's an easy change and it makes sense!